pkg/csi/manager.go

```diff
@@ -70,8 +76,36 @@ func (m *Manager) Run(cfg *config.Config) error {
 		return err
 	}
 
+	nodeID := cfg.NodeID
+
+	ifaces, err := sysfsnet.Interfaces()
+	if err != nil {
+		logrus.WithError(err).Warn("Failed to enumerate MAC addresses for VMI discovery")
+	}
+
+	name, err := discoverVMIName(virtClient.VirtualMachineInstance(namespace), ifaces)
```

Comment:
Could we 
1. first check with the `cfg.NodeID`.
2. If it fails, then try to discover with whole VMs.

Most cases will be confirmed with the first check.
Because every time checking with whole VMs might have a little bigger overhead.
That would have less overhead if we could be done with checking a single VM.

Reply:
:+1: Great idea, I'll add this to the next revision :slightly_smiling_face: